src/order/compactly_generated.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import order.well_founded
+import order.order_iso_nat
+import data.set.finite
+import tactic.tfae
+
+/-!
+# Compactness properties for complete lattices
+
+For complete lattices, there are numerous equivalent ways to express the fact that the relation `>`
+is well-founded. In this file we define three especially-useful characterisations and provide
+proofs that they are indeed equivalent to well-foundedness.
+
+## Main definitions
+ * `is_sup_closed_compact`
+ * `is_Sup_finite_compact`
+ * `is_compact_element`
+ * `is_compactly_generated`
```

Comment:
```suggestion
 * `complete_lattice.is_sup_closed_compact`
 * `complete_lattice.is_Sup_finite_compact`
 * `complete_lattice.is_compact_element`
 * `complete_lattice.is_compactly_generated`
```

Reply:
Personally, in informal English, I would say "joins above" or "join gets above", but I don't really know if this is common. I'm perfectly happy to accept these wording changes though.